File: solutions/1701-1800/1768-merge-strings-alternately/solution.cpp
#include "solution.hpp"

namespace leetcode::p1768 {

std::string solve(const std::string& word1, const std::string& word2) {
    std::string result;
    result.reserve(word1.size() + word2.size());

    std::size_t i = 0;
    while (i < word1.size() || i < word2.size()) {
        if (i < word1.size()) {
            result += word1[i];
        }
        if (i < word2.size()) {
            result += word2[i];
        }
        ++i;
    }

    return result;
}

} // namespace leetcode::p1768

File: solutions/1701-1800/1768-merge-strings-alternately/solution.ml
let solve word1 word2 =
  let len1 = String.length word1 and len2 = String.length word2 in
  let buffer = Buffer.create (len1 + len2) in
  for i = 0 to max len1 len2 - 1 do
    if i < len1 then Buffer.add_char buffer word1.[i];
    if i < len2 then Buffer.add_char buffer word2.[i]
  done;
  Buffer.contents buffer
